### ISSUE TITLE
Attempt#2:  Missing support for LongPaths

### ISSUE THREAD

**Issue by chewiiieee** [microsoft/testfx#637]
#633  This is a real issue in MSTest

Open a DOS BOX, and type:
MKDIR "c:\longPathnames\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\shortpath"
MKDIR "c:\longPathnames\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\longpath"

The following works in my application:
Public Sub ValidateLongPathSupport()
Assert.IsTrue(System.IO.Directory.Exists("c:\longPathnames\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\shortpath"), "testcase broken")
Assert.IsTrue(System.IO.Directory.Exists("c:\longPathnames\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\123456789\longpath"), "missing longpath support")

End Sub

Calling the same function from Test Explorer it now clearly fails.
<TestClass()>
Public Class UnitTestLongPath
<TestMethod()>
Public Sub TestLongPath()
LongPathApp.ValidateLongPathSupport()
End Sub
End Class

**Comment by chewiiieee** [comment]
The above code in my app requires the appropriate manifest and config file which is attached in #633 

Also note the Windows OS Build number when attempting to reproduce as there is an earlier Windows 10 build which accidentally enabled LongPath support, which won't reproduce the behaviour.

**Comment by vagisha-nidhi** [comment]
@chewiiieee 
The code which I mentioned earlier, this failed in .NETFramework 4.6.1. 
```
Dim basicpath As String = "c:\"
Dim LONGPATHSYNTAX As String = "\\?\"
Dim basiclongpath As String = $"{LONGPATHSYNTAX}{basicpath}"
Dim foo = System.IO.Directory.Exists(basiclongpath)
```
Updating the framework version on 4.7.2 helped to no longer reproduce the issue.
You app(AppMain.vb) and test code are behaving exactly the same when run on the latest .Net Framework version. Please check what framework version you are using in your app and test projects when running this.

**Comment by chewiiieee** [comment]
@Vagisha
I logged the issue using .NETFramework 4.8.

Can you run winver.exe and tell me what version of Windows you are using to
reproduce?

**Comment by vagisha-nidhi** [comment]
My OS build version is 18362.418 
I am providing the gifs on how the app main and the test is behaving the same when I run the project provided by you on my machine.
I have simply created a test method in your unit test project and checking if long path support is there by checking if basic long path exists. This returns true in the first gif.
```
Dim basicpath As String = "c:\"
Dim LONGPATHSYNTAX As String = "\\?\"
Dim basiclongpath As String = $"{LONGPATHSYNTAX}{basicpath}"
Dim foo = System.IO.Directory.Exists(basiclongpath)
```
![net48video](https://user-images.githubusercontent.com/13175100/67477901-a36f4e80-f678-11e9-8573-e0ef57224c48.gif)

Next I ran app main project directly without invoking from test code and this fails.
![appmainFailedVideo](https://user-images.githubusercontent.com/13175100/67478073-fc3ee700-f678-11e9-9461-88723ba9ec70.gif)
This is exactly how the test code is behaving as well. It shows the error message "Long filepath support without "\\?\" is broken" even when running directly from AppMain.vb.
The appropriate config and manifest file are present in the project.
When the variable bUseLongPathsyntax is true, all long path checks are passing. This only fails when bUseLongPathsyntax is false. Which seems like the default behavior.
